structure_tests/drivers/docker_driver.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drivers
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"context"
+
+	"bufio"
+
+	"regexp"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/structure_tests/types/unversioned"
+	"github.com/fsouza/go-dockerclient"
+)
+
+type DockerDriver struct {
+	cli           docker.Client
+	originalImage string
+	currentImage  string
+}
+
+func (d *DockerDriver) Info() string {
+	return fmt.Sprintf("DockerDriver:\nOriginalImage: %s\ncurrentImage: %s\ncli: %s\n",
+		d.originalImage, d.currentImage, d.cli)
+}
+
+func NewDockerDriver(image string) *DockerDriver {
+	newCli, err := docker.NewClientFromEnv()
+	if err != nil {
+		panic(err)
+	}
+	return &DockerDriver{
+		originalImage: image,
+		currentImage:  image,
+		cli:           *newCli,
+	}
+}
+
+func (d *DockerDriver) ProcessCommand(t *testing.T, envVars []unversioned.EnvVar, fullCommand []string,
+	shellMode bool, checkOutput bool) (string, string, int) {
+
+	if len(fullCommand) == 0 {
+		t.Logf("empty command provided: skipping...")
+		return "", "", -1
+	}
+	var env []string
+	for _, envVar := range envVars {
+		env = append(env, envVar.Key+"="+envVar.Value)
+	}
+	stdout, stderr, exitCode := d.exec(t, env, fullCommand)
+
+	if stdout != "" {
+		t.Logf("stdout: %s", stdout)
+	}
+	if stderr != "" {
+		t.Logf("stderr: %s", stderr)
+	}
+	//reset image for next test
+	d.currentImage = d.originalImage
```

Comment:
Would it make sense to do this at the start, instead of the end?

Reply:
it needs to be at the end, before we process the command setup commands have been run so the current image is modified

Reply:
Do you still need this line with the new design?

Reply:
nope. removed